main.cpp

```diff
@@ -758,6 +572,81 @@ void part6()
     std::cout << "it3 after: " << it3 << std::endl;
     std::cout << "---------------------\n" << std::endl;    
 }
+*/
+
+void part7()
+{
+    Numeric ft3(3.0f);
+    Numeric dt3(4.0);
+    Numeric it3(5);
+    
+    std::cout << "Calling Numeric<float>::apply() using a lambda (adds 7.0f) and Numeric<float> as return type:" << std::endl;
+    std::cout << "ft3 before: " << ft3 << std::endl;
+
+    {
+        using NumType = decltype(ft3)::Type;
+        using ReturnType = decltype(ft3);
+
+
+        ft3.apply( [&ft3](std::unique_ptr< NumType >& fNum) -> ReturnType&
+        {
+            *fNum += 7.0f;
+            std::cout << "Address of caller is " << &ft3 << std::endl; //10)
+            return ft3;
+        } );
```

Comment:
remove these std::cout messages.  
they cause your output to not match the expected output.
- [ ] revise